frontend/server/utils.ts

```diff
@@ -65,6 +66,23 @@ export function loadJSON<T>(filepath?: string, defaultValue?: T): T | undefined
   }
 }
 
+/**
+ * prune subMaybePrunedPath to relativePath, then concat with rootFixedPath.
+ */
+export function pruneAndConcatPath(
```

Comment:
nit: Can we give this a more meaningful name? Because knowing a function prunes and concats do not help us understand why we want to do these operations.

After thinking for a while, I think `resolveFilePathOnVolume` can be clearer.
I'm thinking about this signature:
```
function resolveFilePathOnVolume({ filePathInVolume, volumeMountPath, volumeMountSubPath }): string
```

note: it's better to use object as arguments because each argument has a name, it brings better readability when calling the method (all arguments have a name next to it).

Reply:
Great!